checks/remotesettings/uptake_error_rate.py

```diff
@@ -47,6 +48,13 @@
       AND event_string_value <> 'up_to_date'
       {version_condition}
       {channel_condition}
+),
+-- Enumerate all the statuses reported by a client for each source
+row_number_by_client_id AS (
```

Comment:
I don't think CTEs have any extra cost in the execution, but I could be wrong.

And if we do the `GROUP BY` on the first query, that means we have to use an aggregation operator (ie. min, max, etc.) for each selected field that is not grouped. I thought it would be less readable, but again I could be wrong